### ISSUE TITLE
dynamic sql, false positives unused variables

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#3266]
see https://github.com/abap2UI5/abap2UI5/pull/1000

**Comment by larshp** [comment]
reproduced as 

```abap
DATA from TYPE string.
DATA fields TYPE string.
DATA where TYPE string.
DATA result TYPE STANDARD TABLE OF string WITH DEFAULT KEY.

SELECT
  FROM (from)
  FIELDS (fields)
  FOR ALL ENTRIES IN @result
  WHERE (where)
  INTO TABLE @result.
```
